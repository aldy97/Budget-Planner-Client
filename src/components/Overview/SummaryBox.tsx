import React from "react";
import styled from "styled-components";

const StyledBox = styled.div<Box>`
  padding: 80px;
  border-radius: 10px;
  margin: 32px;
  text-align: center;
  flex: 1;
  font-size: 30px;
  background: ${props =>
    props.type === "income"
      ? "linear-gradient(rgb(66, 161, 236), rgb(0, 112, 201))"
      : "linear-gradient(#ffa940,#d46b08)"};
  color: #fff;
`;

interface Box {
  type?: "expense" | "income";
  amount?: string;
}

function SummaryBox({ type, amount }: Box): JSX.Element {
  return (
    <StyledBox type={type}>
      <div data-test="type">
        {type === "expense" ? "Monthly Expense" : "Monthly Income"}:
      </div>
      <div data-test="amount">${amount}</div>
    </StyledBox>
  );
}

export default SummaryBox;
