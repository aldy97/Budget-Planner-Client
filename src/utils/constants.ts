export const COLORS = {
  THEMEBLUE: "#1890ff",
  UNSELECTEDOPTION: "#000000a6",
  ORANGE: "#ffa940",
};

export const ExpenseCategories = [
  "Commute",
  "Meals",
  "Shoppings",
  "Grocery",
  "Veggies",
  "Fruits",
  "Snacks",
  "Sports",
  "Entertainment",
  "Phone Plan",
  "Make up",
  "Housing",
  "Clothing",
  "Children",
  "Elders",
  "Social",
  "Travelling",
  "Wine",
  "Digitals",
  "Car",
  "Medical Care",
  "Books",
  "Learning",
  "Pets",
  "Gifts",
  "Office",
  "Maintainance",
  "Donation",
];

export const IncomeCategories = [
  "Salary",
  "Part-time",
  "Investment",
  "Gifts",
  "Others",
];

export const URL = "http://localhost:7001";
