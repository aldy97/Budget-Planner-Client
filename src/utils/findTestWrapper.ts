import { ReactWrapper } from "enzyme";

export const findTestWrapper = (
  wrapper: ReactWrapper,
  tag: string
): ReactWrapper => {
  return wrapper.find(`[data-test="${tag}"]`);
};
