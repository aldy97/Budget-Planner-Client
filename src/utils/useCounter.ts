import { useEffect, useState } from "react";

export default (): number => {
  const [count, setCount] = useState(0);

  useEffect(() => {
    setTimeout(() => {
      setCount(count + 1);
    }, 1000);
  }, [count]);

  return count;
};
