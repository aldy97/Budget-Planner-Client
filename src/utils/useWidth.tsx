import { useState, useEffect } from "react";

const useInnerWidth = (): number => {
  const [width, setWidth] = useState<number>(window.innerWidth);

  useEffect(() => {
    const handleResize = () => {
      setWidth(window.innerWidth);
    };
    window.addEventListener("resize", handleResize);
  }, [width]);

  return width;
};

export default useInnerWidth;
