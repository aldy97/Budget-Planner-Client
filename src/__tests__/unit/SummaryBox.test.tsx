import React from "react";
import { mount } from "enzyme";
import SummaryBox from "../../components/Overview/SummaryBox";
import { findTestWrapper } from "../../utils/findTestWrapper";

describe("SummaryBox", () => {
  const wrapper = mount(<SummaryBox type="expense" amount={12}></SummaryBox>);

  it("can render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("can render whether it is expense or income box", () => {
    const type = findTestWrapper(wrapper, "type");
    const amount = findTestWrapper(wrapper, "amount");
    expect(type.text()).toEqual("Monthly Expense:");
    expect(amount.text()).toEqual("$12");
  });
});
