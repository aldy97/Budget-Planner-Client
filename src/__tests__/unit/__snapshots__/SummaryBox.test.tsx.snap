// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryBox can render without crashing 1`] = `ReactWrapper {}`;
